vrf: Merge table ID from current

cathay4t opened the pull request:
When desire state does not contain table ID for existing VRF interface,
we merge it from current.

In code:
 * expand `pre_edit_cleanup()` to accept `current: Option<&Interface>`.
 * expand `pre_verify_cleanup()` to accept
   `pre_apply_current: Option<&Interface>`

Integration test case included.

ffmancera commented on rust/src/lib/ifaces/vrf.rs:
```diff
@@ -44,14 +44,59 @@ impl VrfInterface {
     }
 
-    pub(crate) fn pre_verify_cleanup(&mut self) {
+    // Merge table ID from current is desired table ID is 0
```
nit: `s/is/if`  thanks!

ffmancera commented:
Really small nit, thank you!

cathay4t committed:
vrf: Merge table ID from current

When desire state does not contain table ID for existing VRF interface,
we merge it from current.

In code:
 * expand `pre_edit_cleanup()` to accept `current: Option<&Interface>`.
 * expand `pre_verify_cleanup()` to accept
   `pre_apply_current: Option<&Interface>`

Integration test case included.

Signed-off-by: Gris Ge <fge@redhat.com>
